kuma/payments/views.py

```diff
@@ -48,7 +49,10 @@ def recurring_payment_management(request):
         context["cancel_request"] = True
         cancel_success = False
         try:
-            cancel_stripe_customer_subscription(request.user.stripe_customer_id)
+            for subscription in cancel_stripe_customer_subscription(
+                request.user.stripe_customer_id
+            ):
+                UserSubscription.set_canceled(request.user, subscription.id)
```

Comment:
Note my boldness of not swallowing errors. This operation is so "low-level" that my reasoning is that if the MySQL writes fail, then almost all hope is lost.